daisy/resource.go

```diff
@@ -47,17 +47,22 @@ type Resource struct {
 	users            []*Step
 }
 
-func (r *Resource) populate(ctx context.Context, s *Step, name, zone string) (string, string, dErr) {
-	errs := r.populateHelper(ctx, s, name)
+func (r *Resource) populateWithGlobal(ctx context.Context, s *Step, name string) (string, dErr) {
+	errs := r._populateHelper(ctx, s, name)
+	return r.RealName, errs
+}
+
+func (r *Resource) populateWithZone(ctx context.Context, s *Step, name, zone string) (string, string, dErr) {
+	errs := r._populateHelper(ctx, s, name)
 	return r.RealName, strOr(zone, s.w.Zone), errs
 }
 
 func (r *Resource) populateWithRegion(ctx context.Context, s *Step, name, region string) (string, string, dErr) {
-	errs := r.populateHelper(ctx, s, name)
-	return r.RealName, strOr(region, s.w.Zone[:len(s.w.Zone)-2]), errs
+	errs := r._populateHelper(ctx, s, name)
+	return r.RealName, strOr(region, getRegionFromZone(s.w.Zone)), errs
 }
 
-func (r *Resource) populateHelper(ctx context.Context, s *Step, name string) dErr {
+func (r *Resource) _populateHelper(ctx context.Context, s *Step, name string) dErr {
```

Comment:
Don't start function names with an underscore

Reply:
done

Reply:
is there a better way of marking that interface as being private? In python the underscores are normally used for protected and private scope visibilities.

Reply:
lowercase is private, Uppercase is public. Also its actually enforces unlike in python.